### PR TITLE
Adding mapping Shiny app

### PR DESCRIPTION
**Pull request by gareth-j**
This adds the basics of a tutorial for the `plot_map` function and what we want it to eventually do. 
For discussion see #14.

### CODE REVIEW

**Comment by gareth-j** [comment]
So there's now a working interactive plotting tool in the form of `plot_interactive_map`. This takes raster and optionally, polygon data. See this [this short tutorial](https://github.com/4DModeller/fdmr/blob/Iss14/MapVignette/vignettes/interactive_mapping.Rmd) for how to use it.

File: R/shiny_mapper.R [hunk 1]
```diff
@@ -3,46 +3,51 @@
...
-    # Define UI for application that draws a histogram
-    ui <- shiny::fillPage(
-        shiny::tags$head(
-            shiny::tags$style(shiny::HTML("
+raster_mapping_app <- function(raster_data = NULL, polygon_data = NULL, date_format = NULL, palette = NULL) {
+  # TODO - can we check the projection of the data?
+  # TODO - check projection of the polygon data? Or require users to do this?
+
+  # Remove any letters from the names of the RasterLayers
+  cleaned_names <- stringr::str_replace_all(names(raster_data), pattern = "[A-Za-z]", replacement = "")
```
Comment (ritarasteiro) [inline R/shiny_mapper.R:17]: Is stringr in the Description file?

File: R/shiny_mapper.R [hunk 2]
```diff
@@ -3,46 +3,51 @@
...
+  # Create Date objects
+  date_strings <- lubridate::as_date(cleaned_names, format = date_format)
+  date_strings <- base::sort(date_strings)
+
+  if (any(is.na(date_strings))) {
+    stop(paste(
```
Comment (ritarasteiro) [inline R/shiny_mapper.R:23]: Don't need `paste()` here

**Review by ritarasteiro — APPROVED** [review]
Hi Gareth, 

I am not very familiar with Shiny, but I think it looks good!
I have just some minor comments, the majority related with remote packages and imports.

File: R/util_packages.R [hunk 3]
```diff
@@ -2,12 +2,23 @@
...
 #'
-#' @param name Package name
+#' @param packages Package name
 #'
 #' @return NULL
 #' @keywords internal
-require_package <- function(pkg_name) {
-  if (!requireNamespace(pkg_name, quietly = TRUE)) {
-    stop(paste("The package", pkg_name, "is required by this function. Please install."))
+require_packages <- function(packages) {
```
[the remark this answers is not in the document]
Comment (gareth-j) [reply R/util_packages.R:9]: Thanks, I've stopped using it for now.

File: R/shiny_mapper.R (hunk 1 above)
Comment (gareth-j) [reply R/shiny_mapper.R:17]: I've just added it, thanks for catching that.

**Pull request merged**